subsys/net/lib/http/http_client.c

```diff
@@ -29,12 +29,32 @@ LOG_MODULE_REGISTER(net_http_client, CONFIG_NET_HTTP_LOG_LEVEL);
 #define HTTP_CONTENT_LEN_SIZE 11
 #define MAX_SEND_BUF_LEN 192
 
-static int sendall(int sock, const void *buf, size_t len)
+static int sendall(int sock, const void *buf, size_t len, int32_t *timeout)
 {
+	int64_t timestamp = k_uptime_get();
```

Comment:
There is the `sys_timepoint` API we could use here, it is used in the system for this purposes (to calculate the timeouts)

Reply:
@jukkar I just tried to stay consistent with what was there, but I guess you are right, I'll check out `sys_timepoint` API and will replace the calls accordingly.

Reply:
@jukkar You are right, it's indeed cleaner.
I just passed down a `k_timepoint_t` by which the function should add or timeout.

The only downside is that I had to convert then `k_timeout` to ms, since `zsock_poll` expects ms (which is correct, since posix `poll` expects milliseconds).